src/main/scala/viper/gobra/reporting/VerifierError.scala

```diff
@@ -266,6 +266,16 @@ case class ChannelMakePreconditionError(info: Source.Verifier.Info) extends Veri
   override def localMessage: String = s"The provided length to ${info.origin.tag.trim} might be negative"
 }
 
+case class RangeVariableMightNotExistError(info: Source.Verifier.Info)(varName: String, rangeExpr: String) extends VerificationError {
+  override def localId: String = "range_variable_might_not_exist"
+  override def localMessage: String = s"Range variable '$varName' might not have been declared because the length of the expression '$rangeExpr' might be 0"
+}
+
+case class NoPermissionToRangeExpressionError(info: Source.Verifier.Info) extends VerificationError {
+  override def localId: String = "no_permission_to_range_expression"
+  override def localMessage: String = s"Generated invariant for the loop with range clause cannot hold. Missing permissions for range expression"
+}
```

Comment:
I find both error messages a bit confusing. I would go with something along the lines of "Might not have read permissions to range expression." since the user does not have to know that the generated invariant causes this, and "The length of the range expression might be 0." since the current statement about the existence of ranged variables seems missleading.